strawberryfields/backends/gaussianbackend/backend.py

```diff
@@ -201,16 +201,19 @@ def measure_fock(self, modes, shots=1, select=None):
                           "Conditional state after Fock measurement has not been updated.")
 
         mu = self.circuit.mean
+        mean = self.circuit.smeanxp()
         cov = self.circuit.scovmatxp()
-        # check we are sampling from a gaussian state with zero mean
-        if not allclose(mu, zeros_like(mu)):
-            raise NotImplementedError("PNR measurement is only supported for "
-                                      "Gaussian states with zero mean")
+
         x_idxs = array(modes)
         p_idxs = x_idxs + len(mu)
         modes_idxs = concatenate([x_idxs, p_idxs])
         reduced_cov = cov[ix_(modes_idxs, modes_idxs)]
-        samples = hafnian_sample_state(reduced_cov, shots)
+        reduced_mean = mean[modes_idxs]
+        # check we are sampling from a gaussian state with zero mean
+        if allclose(mu, zeros_like(mu)):
+            samples = hafnian_sample_state(reduced_cov, shots)
+        else:
+            samples = hafnian_sample_state(reduced_cov, shots, mean=reduced_mean)
```

Comment:
:fire:

Reply:
Thanks! As for the tests, should they be frontend or backend?

Reply:
Ideally both; one in `tests/backend/` to test the above implementation directly, and another in `tests/integration`, to ensure that it works correctly when accessed from the frontend

Reply:
Apparently sf does not know of the latest version the walrus... and because of this the tests are failing in travis...

Reply:
Ah. We will need to release a new version of TheWalrus before the tests will be able to pass here. This should be fine, I think there are more than enough new features in TheWalrus to warrant a new release.